### PR TITLE
fix: bugs caused by Select upgrade to antd4.x

### PR DESCRIPTION
**Pull request by hujiahao-hjh**
## What type of PR is this?

- [ ] Feature
- [x] Bugfix
- [ ] Test
- [ ] Documentation
- [ ] Refactor
- [ ] Style
- [ ] Chore

## Which issue(s) this PR fixes:

Fixes # bugs caused by Select upgrade to antd4.x

## Does this PR introduce a user interface change?

- [ ] Yes(screenshot is required)
- [x] No

### CODE REVIEW

File: core/src/nusi/fixed-select.tsx [hunk 1]
```diff
@@ -17,9 +17,10 @@ import { get } from 'lodash';
 
 const { Option, OptGroup } = Select;
 const FixedSelect = React.forwardRef((props, ref) => {
+  const {options: propsOptions, ...rest} = props;
```
Comment (McDaddy) [inline core/src/nusi/fixed-select.tsx:20]: prettier not working?

File: core/src/nusi/fixed-select.tsx [hunk 2]
```diff
@@ -17,9 +17,10 @@ import { get } from 'lodash';
...
   const options = props.children || get(props, 'options');
   return (
-    <Select ref={ref} getPopupContainer={(triggerNode) => triggerNode.parentElement as HTMLElement} {...props}>
+    <Select ref={ref} getPopupContainer={(triggerNode) => triggerNode.parentElement as HTMLElement} {...rest}>
```
Comment (McDaddy) [inline core/src/nusi/fixed-select.tsx:23]: u should consider two scenarios
1. user pass options like `[{ label: 'A', value: 'A' }]`,
2. user pass options as React comp array `[<Option .../>]`

File: core/src/nusi/fixed-select.tsx (hunk 1 above)
Comment (hujiahao-hjh) [reply core/src/nusi/fixed-select.tsx]: There's something wrong with my prettier and it's not working.

File: core/src/nusi/fixed-select.tsx [hunk 3]
```diff
@@ -17,9 +17,13 @@ import { get } from 'lodash';
...
 const FixedSelect = React.forwardRef((props, ref) => {
-  const options = props.children || get(props, 'options');
+  const { options: propsOptions, ...rest } = props;
+  
+  //Determine if the child elements of props.options is components
+  const isComponent = props.options?.find(item => item.type?.name === 'Option');
+  const options = isComponent ? props.children || get(props, 'options') : '';
   return (
-    <Select ref={ref} getPopupContainer={(triggerNode) => triggerNode.parentElement as HTMLElement} {...props}>
+    <Select ref={ref} getPopupContainer={(triggerNode) => triggerNode.parentElement as HTMLElement} {...(isComponent && rest || props)}>
```
Comment (McDaddy) [inline core/src/nusi/fixed-select.tsx:26]: prettier not working?

Comment (McDaddy) [inline core/src/nusi/fixed-select.tsx:26]: ambiguous condition check `A & B || C`， use bracket to clarify it

File: core/src/nusi/fixed-select.tsx [hunk 4]
```diff
@@ -17,9 +17,13 @@ import { get } from 'lodash';
 
 const { Option, OptGroup } = Select;
 const FixedSelect = React.forwardRef((props, ref) => {
-  const options = props.children || get(props, 'options');
+  const { options: propsOptions, ...rest } = props;
```
Comment (McDaddy) [inline core/src/nusi/fixed-select.tsx:20]: won't u get lint error? type of props doesn't contain options

Comment (McDaddy) [inline core/src/nusi/fixed-select.tsx:20]: u defined a variable names `propsOptions`, but never use it?

File: core/src/nusi/fixed-select.tsx [hunk 5]
```diff
@@ -17,9 +17,13 @@ import { get } from 'lodash';
...
+  
+  //Determine if the child elements of props.options is components
+  const isComponent = props.options?.find(item => item.type?.name === 'Option');
```
Comment (McDaddy) [inline core/src/nusi/fixed-select.tsx:23]: use `React.isValidElement` instead

File: core/src/nusi/fixed-select.tsx [hunk 6]
```diff
@@ -17,9 +17,13 @@ import { get } from 'lodash';
...
+  const options = isComponent ? props.children || get(props, 'options') : '';
```
Comment (McDaddy) [inline core/src/nusi/fixed-select.tsx:24]: if user pass `props.children`, will u still need to check whether its React component?

Comment (McDaddy) [inline core/src/nusi/fixed-select.tsx:24]: `null` instead of empty string

File: core/src/nusi/fixed-select.tsx (hunk 4 above)
Comment (hujiahao-hjh) [reply core/src/nusi/fixed-select.tsx:20]: Just to pick all the properties in props except the options, and the alias name is used so as not to conflict with options in the following code.

Comment (hujiahao-hjh) [reply core/src/nusi/fixed-select.tsx:20]: Options is a read-only property and cannot delete.

File: core/src/nusi/fixed-select.tsx (hunk 6 above)
Comment (hujiahao-hjh) [reply core/src/nusi/fixed-select.tsx:24]: I think it's not necessary, when users pass props.children, it should be placed in <Select></Select>, users should not be passing Object or Array and so on.

File: core/src/nusi/fixed-select.tsx (hunk 3 above)
Comment (hujiahao-hjh) [reply core/src/nusi/fixed-select.tsx]: I thought my prettier works when I save but it wasn't.

**Pull request merged**